### PR TITLE
Add front-end performance tracking

### PR DESCRIPTION
**Pull request by federicoweber**
### Purpose
To add front-end performance tracking on Datadog.
I hope this is useful :smile:.

### Notes
The middleware is just a wrapper around @bufferapp/chronos APIs and exposes Chronos methods via action creators.

The basic functionality is to automatically track all async data fetch requests. 

Also you can easily use it to track any kind of events you are interested in with:
```
store.dispatch(performanceActions.startMeasure({ name: 'foo' }));
store.dispatch(performanceActions.stopMeasure({ name: 'foo' }));
```
or measuring any event from navigation start  with:
```
dispatch(performanceActions.measureFromNavigationStart({ name: 'foo' }));
```

The data sent to Datadog has the following structure:
```
{
    "duration":745.6550000000007,
    "name":"buffer.perf.queuedPostsFetch",
    "navigationStart":1512788235288,
    "start_time":15677.205,
    "tags":["app:publish"]
}
```

_duration is in floating ms_.

#### Staging Deployment

_This repo is CI/CD enabled and staging deployment should be available at:
https://<branch_name>.publish.buffer.com :smile:_

### CODE REVIEW

File: packages/server/rpc/performanceTracking/index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+module.exports = method(
+  'performance',
+  'track performance',
+  ({ data }) => {
+    if (!data.tags) data.tags = [];
+    data.tags.push('app:publish');
+    data.name = `buffer.perf.${data.name}`;
+
+    dogstatsd.histogram(
+      data.name,
```
Comment (hharnisc) [inline packages/server/rpc/performanceTracking/index.js]: Might be good to check that `name` and `duration` exist here, that way we don't get any strange data in datadog :smile:

File: packages/server/rpc/performanceTracking/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const { method } = require('@bufferapp/micro-rpc');
```
Comment (hharnisc) [inline packages/server/rpc/performanceTracking/index.js]: there's a `createError` function on micro rpc that would be handy to throw when input looks off:

```js
throw createError({ 
  message: 'please specify a data.name', // a message with something describing what is missing
  statusCode: 400 // optional default is 400 
})
```

**Comment by federicoweber** [comment]
Thanks for the review @hharnisc and great point about data validation :smile:.

**Comment by hamstu** [comment]
Yep, thanks @hharnisc, and @federicoweber for making this PR! I'm happy to work on this sometime in Baloo. Just haven't triaged it yet.

@federicoweber Do you have some time to sync on how you're using this in Analyze? I'd love to know what you're tracking. (Just to be clear, this doesn't actually track anything right? It's just the framework so we _can_ track.)

**Comment by federicoweber** [comment]
Hey @hamstu, happy to sync on this one :smile:.

Once merged in, the middleware will automatically track all RPC requests. Specifically, it measures how long it takes from `FETCH_START` to `FETCH_SUCCESS|FETCH_FAIL`.

**Comment by federicoweber** [comment]
Going to merge this in :smile:

**Pull request merged**